pkg/zk/generators.go

```diff
@@ -175,13 +174,54 @@ func makeZkPodSpec(z *v1beta1.ZookeeperCluster, volumes []v1.Volume) v1.PodSpec
 		Affinity:   z.Spec.Pod.Affinity,
 		Volumes:    append(z.Spec.Volumes, volumes...),
 	}
-	if reflect.DeepEqual(v1.PodSecurityContext{}, z.Spec.Pod.SecurityContext) {
+	if z.Spec.Pod.SecurityContext != nil {
 		podSpec.SecurityContext = z.Spec.Pod.SecurityContext
 	}
 	podSpec.NodeSelector = z.Spec.Pod.NodeSelector
 	podSpec.Tolerations = z.Spec.Pod.Tolerations
 	podSpec.TerminationGracePeriodSeconds = &z.Spec.Pod.TerminationGracePeriodSeconds
 	podSpec.ServiceAccountName = z.Spec.Pod.ServiceAccountName
+
+	if z.Spec.VolumePermissions == true && z.Spec.Pod.SecurityContext != nil {
+		var securityContext *v1.SecurityContext
+		var initContainerName string
+		if z.Spec.InitContainers == nil || z.Spec.InitContainers[0].SecurityContext == nil {
+			securityContext = &v1.SecurityContext{}
+			initContainerName = "initcontainer"
+		} else {
+			initContainerName = z.Spec.InitContainers[0].Name
+			securityContext = z.Spec.InitContainers[0].SecurityContext.DeepCopy()
+
+		}
+		var arg string
+		if podSpec.SecurityContext == nil || podSpec.SecurityContext.RunAsUser == nil {
```

Comment:
I don't think `podSpec.SecurityContext` could be nil at this point.  `podSpec.SecurityContext.RunAsUser` could though.

Reply:
Done